### PR TITLE
Use result.Result in more places.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/pulumi/pulumi/pull/2573

### CODE REVIEW

File: pkg/resource/deploy/providers/registry.go [hunk 1]
```diff
@@ -191,9 +192,9 @@ func (r *Registry) CheckConfig(olds, news resource.PropertyMap) (resource.Proper
 }
 
 // DiffConfig checks what impacts a hypothetical change to this provider's configuration will have on the provider.
-func (r *Registry) DiffConfig(olds, news resource.PropertyMap) (plugin.DiffResult, error) {
+func (r *Registry) DiffConfig(olds, news resource.PropertyMap) (plugin.DiffResult, result.Result) {
 	contract.Fail()
-	return plugin.DiffResult{}, errors.New("the provider registry is not configurable")
+	return plugin.DiffResult{}, result.FromError(errors.New("the provider registry is not configurable"))
```
Comment (CyrusNajmabadi) [inline pkg/resource/deploy/providers/registry.go]: this is an actual error (as per the contract.Fail above).

File: pkg/resource/plugin/provider.go [hunk 2]
```diff
@@ -41,7 +42,7 @@ type Provider interface {
 	// CheckConfig validates the configuration for this resource provider.
 	CheckConfig(olds, news resource.PropertyMap) (resource.PropertyMap, []CheckFailure, error)
 	// DiffConfig checks what impacts a hypothetical change to this provider's configuration will have on the provider.
-	DiffConfig(olds, news resource.PropertyMap) (DiffResult, error)
+	DiffConfig(olds, news resource.PropertyMap) (DiffResult, result.Result)
```
Comment (swgillespie) [inline pkg/resource/plugin/provider.go]: I admit, I don't like this - this feels like a layering violation to me. I'm not a fan of mixing and matching result.Result and error as return types for different methods.

I think that we should consider logging everything that comes out of `Diff` and `DiffConfig` and keep it returning an `error`, to keep our code consistent here.

Comment (swgillespie) [reply pkg/resource/plugin/provider.go]: I don't think it's wrong to have sentinel errors like `DiffUnknown` coming out of these functions - at least that way a caller can inspect the error and, if it's not a sentinel error, log a diagnostic and return a Bail.

Comment (ellismg) [reply pkg/resource/plugin/provider.go]: I totally agree with this feedback.

**Comment by CyrusNajmabadi** [comment]
Closing out.

**Pull request closed without merging**